java-checks/src/main/java/org/sonar/java/checks/security/LDAPAuthenticatedConnectionCheck.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.security;
+
+import java.util.Collections;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.java.checks.helpers.JavaPropertiesHelper;
+import org.sonar.java.checks.methods.AbstractMethodDetection;
+import org.sonar.java.matcher.MethodMatcher;
+import org.sonar.java.matcher.TypeCriteria;
+import org.sonar.java.model.LiteralUtils;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+
+@Rule(key = "S4433")
+public class LDAPAuthenticatedConnectionCheck extends AbstractMethodDetection {
+
+  @Override
+  protected List<MethodMatcher> getMethodInvocationMatchers() {
+    return Collections.singletonList(
+      MethodMatcher.create()
+        .typeDefinition(TypeCriteria.is("java.util.Hashtable"))
```

Comment:
This can be more general: .typeDefinition(TypeCriteria.subtypeOf("java.util.Map"))

Reply:
Well, for LDAP, according to the [Oracle docs](https://docs.oracle.com/javase/tutorial/jndi/ldap/authentication.html), an implementation of  [DirContext](https://docs.oracle.com/javase/8/docs/api/javax/naming/directory/DirContext.html) must be used, and both implementations [InitialDirContext](https://docs.oracle.com/javase/8/docs/api/javax/naming/directory/InitialDirContext.html) and [InitialLdapContext](https://docs.oracle.com/javase/8/docs/api/javax/naming/ldap/InitialLdapContext.html) use Hashtable.

Having said that, I agree that it's better practice to use an higher abstraction, so will follow your advice.